src/bundle.rs

```diff
@@ -595,16 +638,61 @@ fn split_decor(
     for (id, prop) in cells.bio_phys.iter() {
         let mut seen = Set::new();
         let mut sm = Vec::new();
-        for d in acc::biophys(
+        let biophys = acc::biophys(
             prop,
             lems,
             ions,
             ihp.get(id).ok_or(nml2_error!("should never happen"))?,
-        )? {
+        )?;
+        let mut non_uniform_args: Map<String, Map<String, acc::MechVariableParameter>> = Map::new();
+        for d in biophys.iter() {
+            match d {
+                Decor::Paint(ref r, Paintable::NonUniformMech { ref name, ns, .. })
+                    if densities.contains(name) =>
+                {
+                    if !seen.contains(r) {
+                        for (k, v) in ns.iter() {
+                            non_uniform_args
+                                .entry(r.to_owned())
+                                .or_insert_with(Map::new)
+                                .insert(format!("{name}_{k}"), v.to_owned());
+                        }
+                    }
+                }
+                _ => (),
+            }
+        }
+        for d in biophys {
             match d {
                 Decor::Paint(r, Paintable::Mech(name, _)) if densities.contains(&name) => {
                     if !seen.contains(&r) {
-                        sm.push(acc::Decor::mechanism(&r, &format!("{id}_{r}"), &Map::new()));
+                        if let Some(ns) = non_uniform_args.get(&r) {
```

Comment:
This also seems like a case for `.entry(&r).(...)`

Reply:
Sets don't support the entry API sadly

Reply:
Ah crap! Thanks for that tidbit, though.